vbench/src/vbench/vbench/request_scheduler.cpp

```diff
@@ -63,7 +63,6 @@ RequestScheduler &
 RequestScheduler::stop()
 {
     _queue.close();
-    _thread.stop();
```

Comment:
Looks like we're still starting `_thread` in `start()`; how is its lifecycle controlled?

Reply:
thread 'start' will start the thread
thread 'stop' does 3 things:
1) sets a flag the thread can observe
2) makes active calls to 'slumber' return immediately
3) makes later calls to 'slumber' return immediately

The idea is that a thread loop can check the stop flag to know when to quit. The other idea is that it uses the slumber function to sleep, which will be turned into a nop function once stop has been called, forcing the thread to wake up and smell the coffee.

The lifetime of the main loop of the scheduler is controlled by the time queue. In the abort case we want to stop as soon as possible, so we close the queue and discard all future requests and force slumber to return so the main loop observes that we are done.

In the normal stop case we want to complete the already scheduled requests, so we close the queue and let the main loop drain it. The only difference now is that slumber will actually sleep between dispatching requests and not busy loop.